pulp_python/tests/functional/api/test_crud_content_unit.py

```diff
@@ -3,14 +3,108 @@
 from requests.exceptions import HTTPError
 
 from pulp_smash import api, config, utils
-from pulp_smash.pulp3.constants import ARTIFACTS_PATH
+from pulp_smash.pulp3.constants import REPO_PATH
 from pulp_smash.pulp3.utils import delete_orphans
 
-from pulp_python.tests.functional.constants import PYTHON_CONTENT_PATH, PYTHON_WHEEL_URL
-from pulp_python.tests.functional.utils import gen_python_package_attrs, skip_if
+from pulp_python.tests.functional.constants import (PYTHON_CONTENT_PATH, PYTHON_UPLOAD_PATH,
+                                                    PYTHON_WHEEL_URL, PYTHON_WHEEL_FILENAME)
+from pulp_python.tests.functional.utils import skip_if
 from pulp_python.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
 
 
+class OneShotUploadTestCase(unittest.TestCase):
+    """
+    Test one-shot upload endpoint
+    """
+
+    @classmethod
+    def setUp(cls):
+        """
+        Create class-wide variable.
+        """
+        cls.cfg = config.get_config()
+        delete_orphans(cls.cfg)
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.test_file = {'file': utils.http_get(PYTHON_WHEEL_URL)}
+
+    @classmethod
+    def tearDown(cls):
+        """
+        Clean class-wide variable.
+        """
+        delete_orphans(cls.cfg)
+
+    def test_01_upload_file_without_repo(self):
+        """
+        1) returns a task
+        2) check task status complete
+        3) check created resource of completed task
+        4) ensure only one and it's a content unit
+        """
+        task_url = self.client.post(PYTHON_UPLOAD_PATH,
+                                    files=self.test_file,
+                                    data={'filename': PYTHON_WHEEL_FILENAME})
+        task = self.client.get(task_url['task'])
+        created_resource = task['created_resources'][0]
+        content_unit = self.client.get(created_resource)
+        new_filename = content_unit['filename']
+        self.assertEqual(new_filename, PYTHON_WHEEL_FILENAME)
+        self.assertEqual(len(task['created_resources']), 1)
+
+    def test_02_upload_file_with_repo(self):
+        """
+        1) returns a task
+        2) check task status complete
+        3) check created resource of completed task
+        4) ensure two and it's a content unit and a repository version
+        5) ?
+        """
+        repo = self.client.post(REPO_PATH, data={'name': 'foo'})
+        self.addCleanup(self.client.delete, repo['_href'])
+        task_url = self.client.post(PYTHON_UPLOAD_PATH,
+                                    files=self.test_file,
+                                    data={'filename': PYTHON_WHEEL_FILENAME,
+                                          'repository': repo['_href']})
+        task = self.client.get(task_url['task'])
+        new_repo_version = task['created_resources'][0]
+        version_content_query = self.client.get(new_repo_version)[
```

Comment:
If this can be split up differently somehow, that would be awesome.  I know it's a nitpick but this is hard to read to me.

Reply:
Would it help to add a comment saying what we're trying to access and that it's deeply nested?

Reply:
How about

```
version_content_query = self.client.get(
    new_repo_version
)['content_summary']['added']['python.python']['href']
```

Reply:
Done. I like that much better.  Honestly, with as large as screens are now, I wish PEP8 would ease the character limit per line a bit more because I spend so much time making things harder to read to get around it.

Reply:
Yup, we've historically used 100 as our limit instead of 80